src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -230,6 +230,60 @@ class CreateTableStatementSegment(
         ],
     )
 
+@mysql_dialect.segmentTableConstraint(replace=True)
+class Segment(BaseSegment):
```

Comment:
```suggestion
@mysql_dialect.segment(replace=True)
class TableConstraintSegment(BaseSegment):
```
as I suggested in the original review

Reply:
Right. No idea what I did there. Should be fixed with the latest commit.